packages/web-runtime/src/components/Topbar/Notifications.vue

```diff
@@ -86,26 +93,29 @@ export default {
           })
           res.json().then((json) => {
             json.ocs.data.forEach((item) => {
-              const path = item.path.slice(0, item.path.lastIndexOf('/') + 1)
-              const absolutePath = this.$route.params.item ? this.$route.params.item : '/'
-              if (path === absolutePath) this.reloadFilesList(path)
+              const currentPath = this.$route.params.item ? `/${this.$route.params.item}` : '/'
+              const { state, path, file_target: fileTarget } = item
+              // accepted federated share
+              if (state === 0 && fileTarget) {
+                let shareRoot = fileTarget.substring(0, fileTarget.lastIndexOf('/') + 1)
+                if (shareRoot !== '/') {
+                  // remove trailing slash
+                  shareRoot = shareRoot.replace(/\/+$/, '')
+                }
+
+                if (shareRoot === currentPath) {
+                  this.loadFolder(`files/${this.user.id}/${currentPath}`)
+                  return
+                }
+              }
+
+              if (path) {
+                const itemPath = path.slice(0, path.lastIndexOf('/') + 1)
+                if (itemPath === currentPath) this.loadFolder(`files/${this.user.id}/${itemPath}`)
+              }
```

Comment:
Not 100% sure if we should include such logic here, or if we should move that to the files app and just trigger an event here.